src/Serokell/Util/Bench.hs

```diff
@@ -9,26 +18,22 @@ module Serokell.Util.Bench
        , perSecond
        ) where
 
-import           Control.Monad.Trans (MonadIO (liftIO))
-import           Data.Text.Buildable (Buildable (build))
-import           Data.Time.Units     (Nanosecond, TimeUnit, convertUnit)
-import           Formatting          (bprint, shown, (%))
-import           System.Clock        (Clock (..), TimeSpec, diffTimeSpec,
-                                      getTime, toNanoSecs)
+import Universum
+
+import Fmt (Buildable (build), (+||), (||+))
+import System.Clock (Clock (..), TimeSpec, diffTimeSpec, getTime, toNanoSecs)
+import Time (Time, ns, toUnit, KnownRat)
 
 -- | Get current wall-clock time as any time unit.
-getWallTime :: (MonadIO m, TimeUnit a) => m a
+getWallTime :: (MonadIO m) => forall unit . KnownRat unit => m (Time unit)
```

Comment:
Type signature of this function is a little bit weird... You don't need to use `Rank2Types` extension, you need to specify all type variables before first constraint, like `forall unit m . ...`.

Reply:
:scream: